lib/auxiliary/sysinfo_parser.rb

```diff
@@ -0,0 +1,186 @@
+        # is still in the key field. Remove it.
+        k = k.chomp(DELIM)
+        case k
+        # network_cards currently are not parsed
+        when 'hotfixs', 'processors'
+          raw_output[k] = parse_hotfixs_or_processors(v)
+        when 'hyperv_requirements'
+          raw_output[k] = parse_hyperv_requirements(v)
+        when 'total_physical_memory', 'available_physical_memory', 'virtual_memory_max_size',
+             'virtual_memory_available', 'virtual_memory_in_use'
+          raw_output["#{k}_mb"] = UtilsParser.convert_to_int(v)
+        else
+          raw_output[k] = v&.strip
+        end
+      end
+
+      raw_output
+    end
+
+    # Turns a list of return-delimited hotfixes or processors
+    # with [x] as a prefix into an array of hotfixes
+    #
+    # Note that if a high number of items exist, the list may cutoff
+    # and this list may not contain all items installed on the system
+    #
+    # IRL, this likely applies to hotfixes more than processors
+    #
+    # Parameters:
+    #   data: (string) Input string
+    def parse_hotfixs_or_processors(data)
+      arr_output = []
+      data.split("\n").each_with_index do |l, i|
+        # skip line that says how many are installed
+        next if i.zero?
+
+        # we have to make sure this is a complete line
+        # as data could cutoff. Make sure the delimiter
+        # exists. Otherwise, skip it.
+        next unless l.include?(DELIM)
+
+        _, v = l.split(DELIM)
+        # discard the number sequence
+        arr_output << v.strip
+      end
+
+      arr_output
+    end
+
+    # Turns a list of key/value settings for hyperv
+    # into an object
+    #
+    # Parameters:
+    #   data: (string) Input string
+    def parse_hyperv_requirements(data)
+      output = {}
+      data.split("\n").each do |l|
+        next unless l.include? DELIM
+
+        k, v = l.split(DELIM)
+        # discard the number sequence
+        output[transform_key(k)] = v.strip
+      end
+
+      output
+    end
+
+    # Finds the first non-whitespace character after the delimiter
+    #
+    # @param line [String] Input string
+    # @param delim [String] The data delimiter
+    # @return [Integer] The position of the first non-whitespace character after the delimiter
+    def get_value_pos(line, delim)
+      fields = line.split(delim, 2)
+      unless fields.length == 2
+        raise(StandardError,
+              "Expected a '#{delim}' delimited field. Actual: #{line} (#{fields})")
+      end
+
+      line.length - fields[1].lstrip.length
+    end
+
+    # Converts a given key to a valid JSON key that plays nice with jq
+    #
+    # @param key [String] Input value
+    # @return [String] The transformed key
+    def transform_key(key)
+      # lowercase and replace spaces with underscores
+      key = key.strip.downcase.gsub(' ', '_')
+
+      # remove invalid key characters
+      ";:!@\#$%^&*()-".each_char { |c| key = key.gsub(c, '') }
+      key
+    end
```

Comment:
I don't think this is necessary.

Reply:
Not sure about this. How to add tests in this case. Some keys contains '()' for example.

Reply:
Probably it's fine even if parentheses are present.